### PR TITLE
🔧(dogwood/3/fun) Complete and split (python/yaml) FUN production settings

### PR DESCRIPTION
**Pull request by rmoch**
Set minimum default values in fun.py settings files to allow
images to run, and make configurable those which need to.
Also move real FUN settings to yaml files which will be used
by Matsuo for deployment while being available here for development.

### CODE REVIEW

File: releases/dogwood/3/fun/config/lms/fun.py [hunk 1]
```diff
@@ -231,8 +227,149 @@ def ensure_directory_exists(directory):
...
+# See Xblock i18n: http://www.libremente.eu/2017/12/06/edx-translation/
+LOCALIZED_APPS = sorted([p.split("/")[-2] for p in glob(FUN_BASE_ROOT / "*/locale")])
+LOCALE_PATHS = tuple(
+    [FUN_BASE_ROOT / app / "locale" for app in LOCALIZED_APPS]
+    + [
+        BASE_ROOT / "edx-platform/conf/locale",
+        BASE_ROOT / "venvs/edxapp/lib/python2.7/site-packages/proctor_exam/locale",
+        # this should not be required
+        BASE_ROOT / "venvs/edxapp/lib/python2.7/site-packages/django_countries/locale",
+        BASE_ROOT / "venvs/edxapp/src/edx-proctoring/locale",
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/fun.py]: We don't use venvs in production. I'm not sure what are you targeting here?

File: releases/dogwood/3/fun/config/lms/fun.py [hunk 2]
```diff
@@ -231,8 +227,149 @@ def ensure_directory_exists(directory):
...
+
+PAID_COURSE_REGISTRATION_CURRENCY = ["EUR", u"\N{euro sign}"]
+
+EDX_API_KEY = config("EDX_API_KEY", default="")
+
+# A user is verified if he has an approved SoftwareSecurePhotoVerification entry
+# this setting will create a dummy SoftwareSecurePhotoVerification for user in paybox success callback view
+# I think it's better to create a dummy one than to remove verifying process in edX
+FUN_ECOMMERCE_DEBUG_NO_NOTIFICATION = config(
+    "FUN_ECOMMERCE_DEBUG_NO_NOTIFICATION", default=False
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/fun.py]:
```suggestion
    "FUN_ECOMMERCE_DEBUG_NO_NOTIFICATION", default=False, formatter=bool
```

File: releases/dogwood/3/fun/config/lms/fun.py [hunk 3]
```diff
@@ -231,8 +227,149 @@ def ensure_directory_exists(directory):
...
+)
+
+# when True this setting add a link in instructor dashbord to analytics insigt service
+ANALYTICS_DASHBOARD_URL = config("ANALYTICS_DASHBOARD_URL", default=False)
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/fun.py]:
```suggestion
ANALYTICS_DASHBOARD_URL = config("ANALYTICS_DASHBOARD_URL", default=False, formatter=bool)
```

File: releases/dogwood/3/fun/config/lms/secrets.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+ANONYMIZATION_KEY: ""
+
+EDX_API_KEY: ""
+
+ECOMMERCE_API_SIGNING_KEY: ""
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/secrets.yml]: carriage return is missing

File: releases/dogwood/3/fun/config/lms/settings.yml [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+TECH_SUPPORT_EMAIL: contact@fun-mooc.fr
+CONTACT_EMAIL: contact@fun-mooc.fr
+BUGS_EMAIL: contact@fun-mooc.fr
+PAYMENT_SUPPORT_EMAIL: paiements@fun-mooc.fr
+PAYMENT_ADMIN: paybox@fun-mooc.fr
+BULK_EMAIL_DEFAULT_FROM_EMAIL: no-reply@fun-mooc.fr
+FAVICON_PATH: fun/images/favicon.ico
+
+PLATFORM_FACEBOOK_ACCOUNT: https://www.facebook.com/france.universite.numerique
+PLATFORM_TWITTER_ACCOUNT: "@funmooc"
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/settings.yml]:
```suggestion
PLATFORM_TWITTER_ACCOUNT: "@FunMooc"
```

File: releases/dogwood/3/fun/config/lms/settings.yml [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+PLATFORM_NAME: FUN
+DEFAULT_FROM_EMAIL: no-reply@fun-mooc.fr
+DEFAULT_FEEDBACK_EMAIL: contact@fun-mooc.fr
+DEFAULT_BULK_FROM_EMAIL: no-reply@fun-mooc.fr
+TECH_SUPPORT_EMAIL: contact@fun-mooc.fr
+CONTACT_EMAIL: contact@fun-mooc.fr
+BUGS_EMAIL: contact@fun-mooc.fr
+PAYMENT_SUPPORT_EMAIL: paiements@fun-mooc.fr
+PAYMENT_ADMIN: paybox@fun-mooc.fr
+BULK_EMAIL_DEFAULT_FROM_EMAIL: no-reply@fun-mooc.fr
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/settings.yml]: Can we sort variables alphabetically?

File: releases/dogwood/3/fun/config/cms/settings.yml [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+PLATFORM_NAME: FUN
+DEFAULT_FROM_EMAIL: no-reply@fun-mooc.fr
+DEFAULT_FEEDBACK_EMAIL: contact@fun-mooc.fr
+DEFAULT_BULK_FROM_EMAIL: no-reply@fun-mooc.fr
+TECH_SUPPORT_EMAIL: contact@fun-mooc.fr
+CONTACT_EMAIL: contact@fun-mooc.fr
+BUGS_EMAIL: contact@fun-mooc.fr
+PAYMENT_SUPPORT_EMAIL: paiements@fun-mooc.fr
+PAYMENT_ADMIN: paybox@fun-mooc.fr
+BULK_EMAIL_DEFAULT_FROM_EMAIL: no-reply@fun-mooc.fr
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/cms/settings.yml]: Can we sort variable names alphabetically?

File: releases/dogwood/3/fun/config/lms/settings.yml [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+PASSWORD_MAX_LENGTH: 30
+PASSWORD_COMPLEXITY:
+  UPPER: 1
+  LOWER: 1
+  DIGITS: 1
+
+# Email
+EMAIL_HOST: "email-smtp.eu-west-1.amazonaws.com"
+EMAIL_PORT: 587
+EMAIL_USE_TLS: True
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/settings.yml]:
```suggestion
EMAIL_USE_TLS: true
```

File: releases/dogwood/3/fun/config/cms/settings.yml [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+PASSWORD_MAX_LENGTH: 30
+PASSWORD_COMPLEXITY:
+  UPPER: 1
+  LOWER: 1
+  DIGITS: 1
+
+# Email
+EMAIL_HOST: "email-smtp.eu-west-1.amazonaws.com"
+EMAIL_PORT: 587
+EMAIL_USE_TLS: True
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/cms/settings.yml]:
```suggestion
EMAIL_USE_TLS: true
```

File: releases/dogwood/3/fun/config/lms/settings.yml [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+      ask_to_send_username: true
+      description: <img src='https://glowbl.com/wp-content/uploads/2017/04/Logo2.png'>
+                   <h1>Accès à la salle de visioconférence Glowbl</h1>
+                   <p>
+                   En cliquant sur ce bouton, vous quittez la plateforme <b>fun-mooc.fr</b>
+                   </p><p>
+                   En accédant à la conférence Glowbl, vous acceptez la transmission de votre nom d’utilisateur à la société Glowbl.
+                   </p><p>
+                   Attention une intervention de votre part, écrite ou vidéo, peut être enregistrée.
+                   </p>
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/settings.yml]: Same remark about using YAML block style.

File: releases/dogwood/3/fun/requirements.txt [hunk 11]
```diff
@@ -16,6 +16,9 @@ xblock-utils2==0.3.0
 # ==== xblocks ====
 configurable_lti_consumer-xblock==1.3.0
 
+# ===== glowbl
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/requirements.txt]: Should be in the `xblocks` section?
```suggestion
```

File: releases/dogwood/3/fun/config/cms/settings.yml [hunk 12]
```diff
@@ -0,0 +1,245 @@
...
+      ask_to_send_username: true
+      description: <img src='https://glowbl.com/wp-content/uploads/2017/04/Logo2.png'>
+                   <h1>Accès à la salle de visioconférence Glowbl</h1>
+                   <p>
+                   En cliquant sur ce bouton, vous quittez la plateforme <b>fun-mooc.fr</b>
+                   </p><p>
+                   En accédant à la conférence Glowbl, vous acceptez la transmission de votre nom d’utilisateur à la société Glowbl.
+                   </p><p>
+                   Attention une intervention de votre part, écrite ou vidéo, peut être enregistrée.
+                   </p>
```
[the remark this answers is not in the document]
Comment (rmoch) [reply releases/dogwood/3/fun/config/cms/settings.yml]: Isn't it already multiline string ? 🤔

File: releases/dogwood/3/fun/config/cms/settings.yml (hunk 7 above)
Comment (rmoch) [reply releases/dogwood/3/fun/config/cms/settings.yml]: why not

File: releases/dogwood/3/fun/config/lms/fun.py [hunk 13]
```diff
@@ -143,6 +162,15 @@
...
 FUN_BIG_LOGO_RELATIVE_PATH = "funsite/images/logos/funmoocfp.png"
 
+# Locale paths
+# Here we rewrite LOCAL_PATHS to give precedence to our applications above edx-platform,
+# then we add xblocks which provide i18n as their are no native mecanism.
+# See Xblock i18n: http://www.libremente.eu/2017/12/06/edx-translation/
+LOCALIZED_APPS = sorted([p.split("/")[-2] for p in glob(FUN_BASE_ROOT / "*/locale")])
+LOCALE_PATHS = [FUN_BASE_ROOT / app / "locale" for app in LOCALIZED_APPS]
+LOCALE_PATHS.append(FUN_BASE_ROOT / "edx-platform/conf/locale")
+LOCALE_PATHS.append(path(pkgutil.get_loader("proctor_exam").filename) / "locale")
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/fun.py]: The CMS corresponding setting is buggy then?

File: releases/dogwood/3/fun/config/lms/fun.py [hunk 14]
```diff
@@ -143,6 +162,15 @@
 FUN_SMALL_LOGO_RELATIVE_PATH = "funsite/images/logos/funmooc173.png"
 FUN_BIG_LOGO_RELATIVE_PATH = "funsite/images/logos/funmoocfp.png"
 
+# Locale paths
+# Here we rewrite LOCAL_PATHS to give precedence to our applications above edx-platform,
+# then we add xblocks which provide i18n as their are no native mecanism.
+# See Xblock i18n: http://www.libremente.eu/2017/12/06/edx-translation/
+LOCALIZED_APPS = sorted([p.split("/")[-2] for p in glob(FUN_BASE_ROOT / "*/locale")])
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/fun.py]: Ditto.

File: releases/dogwood/3/fun/config/cms/fun.py [hunk 15]
```diff
@@ -44,8 +78,14 @@
...
-SUBTITLE_SUPPORTED_LANGUAGES = "fr"
-
 # Generic LTI configuration
 LTI_XBLOCK_CONFIGURATIONS = [{"display_name": "LTI consumer"}]
+
+# Locale paths
+# Here we rewrite LOCAL_PATHS to give precedence to our applications above edx-platform,
+# then we add xblocks which provide i18n as their are no native mecanism.
+# See Xblock i18n: http://www.libremente.eu/2017/12/06/edx-translation/
+LOCALIZED_APPS = sorted([p.split("/")[-2] for p in glob(FUN_BASE_ROOT / "*/locale")])
```
[the remark this answers is not in the document]
Comment (rmoch) [reply releases/dogwood/3/fun/config/cms/fun.py]: good idea

File: releases/dogwood/3/fun/config/cms/docker_run_production.py [hunk 16]
```diff
@@ -22,11 +22,15 @@
     update_module_store_settings,
 )
 
-from .fun import *
+from ..common import *
```
[the remark this answers is not in the document]
Comment (rmoch) [reply releases/dogwood/3/fun/config/cms/docker_run_production.py]: you're right

File: releases/dogwood/3/fun/config/lms/fun.py (hunk 13 above)
Comment (rmoch) [reply releases/dogwood/3/fun/config/lms/fun.py]: indeed

File: releases/dogwood/3/fun/config/lms/fun.py [hunk 17]
```diff
@@ -143,6 +162,16 @@
 FUN_SMALL_LOGO_RELATIVE_PATH = "funsite/images/logos/funmooc173.png"
 FUN_BIG_LOGO_RELATIVE_PATH = "funsite/images/logos/funmoocfp.png"
 
+# Locale paths
+# Here we rewrite LOCAL_PATHS to give precedence to our applications above edx-platform's ones,
+# then we add xblocks which provide translations as there is no native mechanism to handle this
+# See Xblock i18n: http://www.libremente.eu/2017/12/06/edx-translation/
+LOCALIZED_FUN_APPS = ["backoffice", "contact", "course_dashboard", "course_pages",
+    "courses", "fun_api", "fun_certificates", "funsite", "newsfeed", "payment", "universities", "videoproviders"]
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/fun.py]: Use black!

File: releases/dogwood/3/fun/config/lms/settings.yml [hunk 18]
```diff
@@ -0,0 +1,293 @@
...
+        </p><p>
+        Attention une intervention de votre part, écrite ou vidéo, peut être enregistrée.
+        </p>
+      button_text: Accéder à la conférence Glowbl et accepter la transmission de mon nom d'utilisateur
+      launch_target: new_window
+      inline_height: 400
+      lti_id: glowbl
+      launch_url: https://account.glowbl.com/auth/provider/lti
+  # Default LTI consumer
+  - display_name': LTI consumer
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/settings.yml]:
```suggestion
  - display_name: LTI consumer
```

File: releases/dogwood/3/fun/config/lms/settings.yml [hunk 19]
```diff
@@ -0,0 +1,293 @@
...
+
+RAVEN_CONFIG:
+  dsn: ""
+
+# FUN Mongo database (TODO: remove)
+FUN_MONGO_DATABASE: fun
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/config/lms/settings.yml]: Should we remove it then?

**Pull request merged**